staging/src/k8s.io/apiserver/pkg/registry/generic/registry/store.go

```diff
@@ -582,11 +582,16 @@ func (e *Store) Update(ctx context.Context, name string, objInfo rest.UpdatedObj
 					return nil, nil, err
 				}
 			}
+
+			// always return a nil TTL when we have no TTL func so that the
+			// storage interface can distinguish between no TTL, 0 TTL, 1+ TTL
```

Comment:
What's the semantic difference between "no TTL" and "0 TTL"? Might be worth clarifying here as it's not obvious.

Reply:
Oh, I see storage.Interface `GuarenteedUpdate` has a listing of the semantics. Maybe reference that?

Reply:
Yeah the distinction is really important on update, but does not matter for create.